File: backend/app/schemas/recruitment_info.py
from pydantic import BaseModel, field_serializer
from typing import Optional
from datetime import datetime


class RecruitmentInfoBase(BaseModel):
    """招考信息基础字段"""
    title: Optional[str] = None
    exam_type: Optional[str] = None
    area: Optional[str] = None
    province: Optional[str] = None
    city: Optional[str] = None
    district: Optional[str] = None
    publish_date: Optional[datetime] = None
    registration_start: Optional[datetime] = None
    registration_end: Optional[datetime] = None
    exam_date: Optional[datetime] = None
    recruitment_count: Optional[int] = None
    status: Optional[str] = None
    source_url: Optional[str] = None
    content: Optional[str] = None
    ai_summary: Optional[str] = None
    attachments: Optional[str] = None
    tags: Optional[str] = None
    source_site: Optional[str] = None
    source_id: Optional[str] = None


class RecruitmentInfoCreate(RecruitmentInfoBase):
    """创建招考信息"""
    pass


class RecruitmentInfoResponse(RecruitmentInfoBase):
    """招考信息响应"""
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class RecruitmentInfoListResponse(BaseModel):
    """招考信息列表响应"""
    items: list[RecruitmentInfoResponse]
    total: int
    page: int
    page_size: int


class RecruitmentInfoFilterOptions(BaseModel):
    """筛选选项"""
    exam_types: list[str]
    provinces: list[str]
    cities: list[str]
    statuses: list[str]


class CrawlerConfigResponse(BaseModel):
    """爬虫配置响应"""
    id: int
    name: Optional[str] = None
    target_url: Optional[str] = None
    interval_minutes: int = 10
    is_active: bool = True
    session_valid: bool = False
    last_crawl_at: Optional[datetime] = None
    last_crawl_status: Optional[str] = None
    last_error: Optional[str] = None
    total_crawled: int = 0
    ai_enabled: bool = False
    ai_model: Optional[str] = None
    ai_api_key: Optional[str] = None
    ai_base_url: Optional[str] = None
    ai_prompt: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    @field_serializer("ai_api_key")
    def mask_api_key(self, v: Optional[str]) -> Optional[str]:
        if not v:
            return v
        if len(v) <= 8:
            return "****"
        return v[:4] + "****" + v[-4:]

    model_config = {"from_attributes": True}


class CrawlerStatusResponse(BaseModel):
    """爬虫状态响应"""
    scheduler_running: bool = False
    configs: list[CrawlerConfigResponse] = []
    recent_count: int = 0
    today_count: int = 0

File: frontend/src/types/recruitmentInfo.ts
// 招考信息类型定义
export interface RecruitmentInfo {
  id: number;
  source_id: string;
  title: string;
  exam_type: string;
  area: string;
  province: string;
  city: string;
  district: string;
  publish_date: string;
  registration_start: string;
  registration_end: string;
  exam_date: string;
  recruitment_count: number;
  status: string;
  source_url: string;
  content: string;
  ai_summary: string;
  attachments: string; // JSON string of attachment links
  tags: string; // JSON string
  source_site: string;
  created_at: string;
}

export interface RecruitmentInfoListResponse {
  items: RecruitmentInfo[];
  total: number;
  page: number;
  page_size: number;
}

export interface RecruitmentInfoFilters {
  exam_type?: string;
  province?: string;
  city?: string;
  status?: string;
  keyword?: string;
  start_date?: string;
  end_date?: string;
  page?: number;
  page_size?: number;
}

export interface RecruitmentInfoFilterOptions {
  exam_types: string[];
  provinces: string[];
  cities: string[];
  statuses: string[];
}

export interface CrawlerConfig {
  id: number;
  name: string;
  target_url: string;
  interval_minutes: number;
  is_active: boolean;
  session_valid: boolean;
  last_crawl_at: string;
  last_crawl_status: string;
  last_error: string;
  total_crawled: number;
  ai_enabled: boolean;
  ai_model: string;
  ai_api_key: string;
  ai_base_url: string;
  ai_prompt: string;
  created_at: string;
}

export interface CrawlerStatus {
  configs: CrawlerConfig[];
  scheduler_running: boolean;
}
